src/main/java/pages/cartPage/shoppingCartTable/ShoppingCartTableRow.java

```diff
@@ -0,0 +1,199 @@
+package pages.cartPage.shoppingCartTable;
+
+import enums.Currency;
+import lombok.Getter;
+import lombok.Setter;
+import models.ProductModel;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import pages.BasePage;
+import pages.cartPage.BaseTable;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import static engine.drivers.WebDriverFactory.getWebDriverInstance;
+
+@Getter
+@Setter
+public class ShoppingCartTableRow {
+    public By deleteItemButton = By.xpath(".//i[contains(@class, 'fa-trash-o')]/parent::a");
+    public By quantityItemInput = By.xpath(".//input[contains(@id, 'cart_quantity')]");
+    protected BasePage basePage;
+    protected BaseTable baseTable;
+    protected WebDriver driver = getWebDriverInstance();
+
+    public ShoppingCartTableRow() {
+        this.baseTable = new BaseTable("//div[contains(@class, 'product-list')]/table");
```

Comment:
no need for the base table here:) just use xpath as parameter here. You will also be able to skip the inheritance in ShoppingCartTable

Reply:
Right. Deleted

Reply:
The relationship between ```ShoppingCartTable``` and ```ShoppingCartTableRow``` has been corrected as we discussed. The ```baseTable``` object is now only in the ```ShoppingCartTable``` where it is used to return the table header names thanks to the